provider/overrides/terraform/property_override.rb

```diff
@@ -0,0 +1,96 @@
+# Copyright 2017 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'api/object'
+require 'provider/overrides/resources'
+require 'provider/terraform/property_override'
+
+module Provider
+  module Overrides
+    module Terraform
+      # Terraform-specific overrides to api.yaml.
+      class PropertyOverride < Provider::Overrides::PropertyOverride
+        # Collection of fields allowed in the PropertyOverride section for
+        # Terraform. All fields should be `attr_reader :<property>`
+        def self.attributes
+          Provider::Terraform::OverrideFields.attributes
```

Comment:
I'm not sure I understand why we do this vs just including it inline. This looks like the only usage & we have to specify the fields in validate anyways?

Reply:
We need a self.attributes to define the attributes that are accessible to the new overrides. The old way of doing this was some module magic that I still can't understand. Having a method makes it more explicit.

I did this OverrideFields thing so that I could share code between new overrides and old overrides.